### PR TITLE
Switch to Hpack

### PR DESCRIPTION
**Pull request by halogenandtoast**
Usage of hpack allows us to avoid having to write and maintain the
exposed-modules directive in our cabal file. Since Hpack is supported by
stack, we simply need to add the package.yaml file and stack will handle
automatically generating the cabal file.

For more information, please read

https://e.xtendo.org/haskell/hpack

### CODE REVIEW

**Comment by halogenandtoast** [comment]
@cookpad/haskell anyone interested in reviewing this?

File: Mirin.cabal [hunk 1]
```diff
@@ -1,47 +1,55 @@
...
-                     , wai
-                     , resourcet
-                     , resource-pool
+name:           Mirin
+version:        0.1.0.0
+category:       Web
+homepage:       https://github.com/githubuser/Mirin#readme
+author:         Author name here
+maintainer:     example@example.com
+copyright:      2017 Author name here
```
Comment (pauchan) [inline Mirin.cabal]: L8-L11 should be filled in? 🤔

**Review by pauchan — COMMENTED** [review]
This kind of sucks that all that metadata is being stored both in `Mirin.cabal` and in `package.yml` .

**Comment by halogenandtoast** [comment]
@pauchan actually, I should be able to remove the cabal file from the repo since it's autogenerated.

**Comment by halogenandtoast** [comment]
@pauchan ♻️

**Review by pauchan — APPROVED** [review]
A lot of red
I like it!

**Pull request merged**